File: 033_scatter_panel_X_consistent_portrait_linear/R/build_data.R
# build_data.R -- Example 033
set.seed(33); N_SUBJ<-60; TRTS<-c("Placebo","Drug A 10mg","Drug A 25mg"); PANELS<-c("Visit 1","Visit 2","Visit 3")
trt_int<-c(Placebo=0,`Drug A 10mg`=6,`Drug A 25mg`=12); panel_shift<-c(`Visit 1`=0,`Visit 2`=5,`Visit 3`=10)
records<-list(); subj_n<-1L
for(panel in PANELS) for(trt in TRTS) for(i in seq_len(N_SUBJ)){ x<-runif(1,10,80); y<-.65*x+trt_int[trt]+panel_shift[panel]+rnorm(1,0,9)
  records[[length(records)+1]]<-data.frame(subjid=sprintf("S%04d",subj_n),panel=panel,trt=trt,x_val=round(x,1),y_val=round(y,1),stringsAsFactors=FALSE); subj_n<-subj_n+1L}
df<-do.call(rbind,records); write.csv(df,"test_data.csv",row.names=FALSE); message(sprintf("Generated %d rows.",nrow(df)))

File: 032_scatter_panel_X_consistent_landscape_linear/python/build_data.py
"""build_data.py -- Examples 032-034: Scatter panels. Generates panel×trt×subj data."""
import numpy as np, pandas as pd
rng=np.random.default_rng(seed=32)
N_SUBJ=60; TRTS=["Placebo","Drug A 10mg","Drug A 25mg"]; PANELS=["Visit 1","Visit 2","Visit 3"]
trt_int={"Placebo":0,"Drug A 10mg":6,"Drug A 25mg":12}; panel_shift={"Visit 1":0,"Visit 2":5,"Visit 3":10}
records=[]; subj=1
for panel in PANELS:
    for trt in TRTS:
        for _ in range(N_SUBJ):
            x=rng.uniform(10,80); y=0.65*x+trt_int[trt]+panel_shift[panel]+rng.normal(0,9)
            records.append({"subjid":f"S{subj:04d}","panel":panel,"trt":trt,"x_val":round(x,1),"y_val":round(y,1)}); subj+=1
df=pd.DataFrame(records); df.to_csv("test_data.csv",index=False); print(f"Generated {len(df)} rows.")
